shell/platform/glfw/flutter_glfw.cc

```diff
@@ -690,6 +693,123 @@ static bool RunFlutterEngine(
   return true;
 }
 
+const char* GetLocaleStringFromEnvironmentVariables(const char* language,
+                                                    const char* lc_all,
+                                                    const char* lc_messages,
+                                                    const char* lang) {
+  const char* retval;
+  retval = language;
+  if ((retval != NULL) && (retval[0] != '\0')) {
+    return retval;
+  }
+  retval = lc_all;
+  if ((retval != NULL) && (retval[0] != '\0')) {
+    return retval;
+  }
+  retval = lc_messages;
+  if ((retval != NULL) && (retval[0] != '\0')) {
+    return retval;
+  }
+  retval = lang;
+  if ((retval != NULL) && (retval[0] != '\0')) {
+    return retval;
+  }
+
+  return NULL;
+}
+
+const char* GetLocaleStringFromEnvironment() {
+  return GetLocaleStringFromEnvironmentVariables(
+      getenv("LANGUAGE"), getenv("LC_ALL"), getenv("LC_MESSAGES"),
+      getenv("LANG"));
+}
+
+// Parse a locale into its components.
+void ParseLocale(const std::string& locale,
+                 std::string* language,
+                 std::string* territory,
+                 std::string* codeset,
+                 std::string* modifier) {
+  // Locales are in the form "language[_territory][.codeset][@modifier]"
+  std::string::size_type end = locale.size();
+  std::string::size_type modifier_pos = locale.rfind('@');
+  if (modifier_pos != std::string::npos) {
+    *modifier = locale.substr(modifier_pos + 1, end - modifier_pos - 1);
+    end = modifier_pos;
+  }
+
+  std::string::size_type codeset_pos = locale.rfind('.');
+  if (codeset_pos != std::string::npos) {
+    *codeset = locale.substr(codeset_pos + 1, end - codeset_pos - 1);
+    end = codeset_pos;
+  }
+
+  std::string::size_type territory_pos = locale.rfind('_');
+  if (territory_pos != std::string::npos) {
+    *territory = locale.substr(territory_pos + 1, end - territory_pos - 1);
+    end = territory_pos;
+  }
+
+  *language = locale.substr(0, end);
+}
+
+std::vector<std::unique_ptr<FlutterLocale>> GetLocales(
+    const char* locale_string,
+    std::list<std::string>& locale_storage) {
+  if (!locale_string || locale_string[0] == '\0') {
+    locale_string = "C";
```

Comment:
Why `"C"`?

Reply:
I found somewhere in some posix docs about locales that "ISO C says that all programs start by default in the standard ‘C’ locale." And looking around at other things that parse these strings, they use "C" as the fallback.